src/pool/worker.rs

```diff
@@ -39,18 +39,39 @@ where
     }
 
     pub fn run(mut self) {
-        self.runner.start(&mut self.local);
-        while !self.local.core().is_shutdown() {
-            let task = match self.pop() {
-                Some(t) => t,
-                None => continue,
-            };
-            self.runner.handle(&mut self.local, task.task_cell);
+        if !self.local.runnable() {
+            self.local.sleep();
         }
-        self.runner.end(&mut self.local);
+        loop {
+            self.runner.start(&mut self.local);
```

Comment:
Initially, `start` and `end` are only called once. Maybe it is not appropriate to change the semantics. `pause` and `resume` are more suitable here.

Reply:
Do `start` and `end` represent the life cycle of a runner? What I expect is that executing `start` again means that it has nothing to do with the previous runner, which is equivalent to a new thread?

Reply:
`Runner::run` is the only thing a worker thread executes. So the lifecycle of a runner equals to the lifecycle of a thread. Initializations or teardowns related to a thread are in the `start` and `end` method, so they should be called only once.

Reply:
So I think my modification is correct. When a thread finds that it is no longer `runnable`, it means that it needs to be destroyed, but we don’t really destroy it, we just call `end` then park it to another queue (the key is not `QueueCore<T>`'s addr, the key is `QueueCore<T>`'addr | 1), no longer participate in the scheduling of `Remote`. When it is `runnable` again, it already represents a new thread, so it needs to call `start` again.

Reply:
OK, I've got what you mean. But it still makes me feel a bit uncomfortable that the thread is just "pretended to be" destroyed.

It may be surprising to users that the `start` method is called multiple times for a thread. It may cause some problems if it  sets something in the thread-local storage but not clean it in the `end` method, or it depends on that the thread id from the `start` method cannot duplicate.

Maybe a safe approach is to define two more trait methods in the `Runner` trait for the scaled threads.

@BusyJay What do you think?

Reply:
Agree with @sticnarf.

Reply:
Add two method `reentrant_start` and `reetrant_end`, does it make sense?